compiler/hash-tir/src/nodes/mod.rs

```diff
@@ -0,0 +1,9 @@
+pub mod args;
+pub mod data;
+pub mod lits;
+pub mod mods;
+pub mod node;
```

Comment:
Why not put the definitions of this file into `mod.rs`

Reply:
What definitions?

Reply:
The ones in `node.rs`

Reply:
Because it is specifically stuff relevant to the `Node` wrapper that is present around the TIR atoms

Reply:
Yeah, but it is the base for everything in this sub-module - it doesn’t seem useful to put it into its own file.

Reply:
I don't know, it seems cleaner to me to have a separate file for this. It makes it more clear where the "general" node stuff is; the `mod.rs` is just there to declare everything. I will add re-exports so the imports in other places should remain unchanged.

Reply:
Fair enough.